pkgs/top-level/all-packages.nix

```diff
@@ -2366,7 +2366,7 @@ with pkgs;
 
   freetds = callPackage ../development/libraries/freetds { };
 
-  frescobaldi = callPackage ../misc/frescobaldi {};
+  frescobaldi = python3Packages.callPackage ../misc/frescobaldi {};
```

Comment:
Please don't use `pythonPackages.callPackage` for applications so people can override python packages used by the application.

Reply:
@dotlambda I thought the same as @jtojnar, but I'm not entirely sure as I'm not a `nixpkgs` internals expert. I'll have a look at it in the next hours and comment here when I know more :)

Reply:
It's definitely possible, but you'd have to override the entire `propagatedBuildInputs`. Let's say you want to use a beta version of pyqt5, then you can just do the following when not using `pythonPackages.callPackage`:
```nix
frescobaldi.override {
  python = python.override {
    packageOverrides = self: super: {
      poppler-qt5 = super.poppler-qt5.overridePythonAttrs (oldAttrs: rec {
        version = "1.0b1";
        src = oldAttrs.src.override {
          inherit version;
          sha256 = "...";
        };
      });
    };
  };
}
```

Reply:
That is not really unique to python, even if it is somewhat exaggerated by the need for package uniqueness. Perhaps there is a way to reScope a package, or apply overlay locally?